test/tests/caliptra_integration_tests/smoke_test.rs

```diff
@@ -510,6 +510,147 @@ fn smoke_test() {
         .cptra_hw_error_non_fatal()
         .read()
         .mbox_ecc_unc());
+
+    let image2 = caliptra_builder::build_and_sign_image(
```

Comment:
Maybe the ship has sailed, but the smoke test is a bit hard to follow since it does so many things. Would it make sense to factor the checks into helpers to make it a little easier to follow. Something like:

```
// Load firmware

verify_uart(...);
verify_cert_fields(...);

// Update firmware

verify_updated_cert_fields(...);
```

Reply:
I do want to do a cleanup commit at some point, moving boilerplaty stuff like the golden-data checks to helper functions or macros. In general though, I think we want to keep important assertions like "the rt-alias must be signed by the fmc-alias" in the top-level test. It's more important that the assertions are clear and "obviously right at a glance". 

The next major change I want to make to the test is to parameterize the UDS and lifecycle/debug values.

Reply:
Ack. Seems fine clean this up in a follow up. I agree having all the asserts in one place is helpful.